src/components/range-inputs.tsx

```diff
@@ -0,0 +1,303 @@
+        let fromTimeVal = fromTimeRef?.current?.value;
+
+        if (fromDateVal && !fromTimeVal) fromTimeVal = '00:00';
+
+        const toDateVal = toRef?.current?.value;
+        let toTimeVal = toTimeRef?.current?.value;
+
+        if (toDateVal && !toTimeVal) toTimeVal = '00:00';
+
+        return {
+            fromVal: fromDateVal || fromTimeVal ? `${fromDateVal}T${fromTimeVal}` : '',
+            toVal: toDateVal || toTimeVal ? `${toDateVal}T${toTimeVal}` : '',
+        };
+    }
+
+    /**checks if both inputs are empty and disables the submit button */
+    const handleChange = (value: string) => {
+        const formatedValues = type === 'timestamp' ? formatTimeValues()
+            : { fromVal: fromRef?.current?.value || '', toVal: toRef?.current?.value || '' };
+
+        console.log(formatedValues);
+
+        if (!formatedValues.fromVal && !formatedValues.toVal) disableSubmitBtn();
+        else enableSubmitBtn();
+
+        const validatedResult = validateValue(value);
+
+        if (!validatedResult) setError(errorMsgMap[type]);
+
+        if (validatedResult && !error) setError(null);
+    }
+
+    const validateValue = (value: string): boolean => {
+        if (!value) return true;
+
+        if (type === 'int') return INTEGER_REGEXP.test(value);
+
+        if (type === 'float') return FLOAT_REGEXP.test(value);
+
+        /**type is either date or timestamp */
+        const formatString = type === 'date' ? DATE_FORMAT : TIME_FORMAT;
+        const date = momentJS(value, formatString, true);
+        return date.isValid();
+    }
+
+    const rangeCheck = (fromVal: string, toVal: string): boolean => {
+        if (type === 'int') return parseInt(fromVal) < parseInt(toVal);
+
+        if (type === 'float') return parseInt(fromVal) < parseInt(toVal);
+
+        /**type is either date or timestamp */
+        const formatString = type === 'date' ? DATE_FORMAT : TIME_FORMAT;
+        const fromDate = momentJS(fromVal, formatString, true);
+        const toDate = momentJS(toVal, formatString, true);
+        return toDate.diff(fromDate) > 0
+    }
+
+
+    const validateValues = (fromVal: string, toVal: string): string => {
+        const isfromValid = validateValue(fromVal);
+
+        const isToValid = validateValue(toVal);
+
+        if (!isfromValid || !isToValid) return type;
+
+        /**if only one field is non null then don't perform range check */
+        if (!fromVal || !toVal) return 'valid';
+
+        /**both from and to values are now valid so perform range validations */
+        return rangeCheck(fromVal, toVal) ? 'valid' : 'range';
+    }
+
+    const handleSubmit = () => {
+        const formatedValues = type === 'timestamp' ? formatTimeValues()
+            : { fromVal: fromRef?.current?.value || '', toVal: toRef?.current?.value || '' };
+
+        const validatedResult = validateValues(formatedValues.fromVal, formatedValues.toVal);
+
+        console.log('validation result: ', validatedResult);
```

Comment:
Please remove this when debugging is done.

Reply:
once you have tested the functionality I'll remove all the logs